fix: correct tooltip text retrieval in multichain site cell component

fabiobozzo opened the pull request:
## **Description**

Fixes two related bugs in the multichain site cell tooltip component:

### Bug `#1`: TypeError when displaying overflow text
The `getMoreText` variable was defined using `useMemo`, which returns a memoized value (a string), but was incorrectly being called as a function `getMoreText()` instead of being used directly as `getMoreText`.

### Bug `#2`: Redux Context error when hovering over accounts
The tooltip content was failing with "could not find react-redux context value" because `react-tippy` renders tooltip content in a separate React root that doesn't inherit the Redux Provider or I18n Context from the main app. The tooltip content was using:
- `PreferredAvatar` component (which uses `useSelector` internally)
- `useI18nContext()` hook (which depends on React Context)

### Bug `#3`: Dual overflow not handled correctly
When both accounts AND networks exceeded the tooltip limit (>4 each), only the accounts overflow message was shown. The logic returned early after the first condition matched, causing the networks overflow to be silently ignored.


**The fixes:**
1. Changed `{getMoreText()}` to `{getMoreText}` to properly use the memoized string value
2. Replaced `PreferredAvatar` with `AvatarAccount` directly using `AvatarAccountVariant.Jazzicon`
3. Moved i18n translation logic from `TooltipContent` to parent component and pass pre-computed text as prop
4. Split overflow logic into separate `moreAccountsText` and `moreNetworksText` to handle both overflow cases independently


## **Changelog**

CHANGELOG entry: null

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/37134

## **Manual testing steps**

1. Connect a site to more than 4 accounts AND more than 4 networks (e.g., 6 accounts and 6 networks)
2. Navigate to the connected sites view where the multichain site cell is displayed
3. Hover over the avatar group that shows the connected **accounts**
4. Verify the tooltip appears showing:
   - First 4 account names
   - "+2 more accounts" text
5. Hover over the avatar group that shows the connected **networks**
6. Verify the tooltip appears showing:
   - First 4 network names
   - "+2 more networks" text
7. When BOTH have overflow, verify the tooltip shows both overflow messages correctly
8. Check browser console for any JavaScript errors - should be none


## **Screenshots/Recordings**

N/A - This is a runtime error fix that doesn't change visual behavior when working correctly.

### **Before**

- TypeError: f is not a function (in production/minified code)
- Error: could not find react-redux context value when hovering over accounts tooltip

### **After**

Tooltip displays correctly for both accounts and networks without errors

<img width="395" height="597" alt="Screenshot 2025-11-05 at 16 27 42" src="https://github.com/user-attachments/assets/0f46a86c-8027-4464-af05-4ce9c2dde1fe" />
<img width="398" height="596" alt="Screenshot 2025-11-05 at 16 27 47" src="https://github.com/user-attachments/assets/5a1eaa7a-909d-4786-ad03-ac4a541492d2" />


## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I've included tests if applicable
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I've applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.


<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Computes i18n overflow texts in the parent, renders both accounts and networks overflow lines, and replaces PreferredAvatar with design-system AvatarAccount (Jazzicon).
> 
> - **Multichain site tooltip (`multichain-site-cell-tooltip.tsx`)**:
>   - Replace `PreferredAvatar` with `AvatarAccount` (`Jazzicon`) and related imports.
>   - Remove i18n usage from `TooltipContent`; add `moreAccountsText` and `moreNetworksText` props and render each overflow line separately with proper guards and test IDs.
>   - Parent computes `moreAccountsText`/`moreNetworksText` via `useMemo` and passes them to `TooltipContent`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 53979edf0ad978a603cfa128fcaf6209e7198851. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

Commit:
fix: correct tooltip text retrieval in multichain site cell component

getMoreText is defined using useMemo, which returns a memoized value (a string), not a function

Commit:
Merge branch 'main' into fix/MUL-1207-typeerror-f-not-a-function

ccharly reviewed (dismissed):
LGTM (not tested, but the code speaks for itself IMO 😁)

Commit:
fix: update tooltip content in multichain site cell component

Refactor TooltipContent to accept a memoized `moreText` prop for displaying additional account and network information. Replace `PreferredAvatar` with `AvatarAccount` for improved avatar rendering. This enhances the tooltip's functionality and visual consistency.

Commit:
Merge branch 'main' into fix/MUL-1207-typeerror-f-not-a-function

Commit:
refactor: update tooltip props for multichain site cell component

Modified the TooltipContent component to accept new props `moreAccountsText` and `moreNetworksText`, replacing the previous `moreText` prop. This enhances the tooltip's ability to display additional information about accounts and networks more clearly. Updated the logic to conditionally render these texts based on the presence of additional accounts and networks.

Commit:
Merge branch 'main' into fix/MUL-1207-typeerror-f-not-a-function
